Applications/FileIO/Legacy/createSurface.cpp

```diff
@@ -77,39 +79,33 @@ bool createSurface(GeoLib::Polyline const& ply,
         0.0, 0, geo_names, false, false);
     gmsh_io.setPrecision(std::numeric_limits<double>::digits10);
 
-    char file_base_name_c[L_tmpnam];
-    if (! std::tmpnam(file_base_name_c))
-    {
-       OGS_FATAL("Could not create unique file name.");
-    }
-    std::string const file_base_name(file_base_name_c);
-    gmsh_io.writeToFile(file_base_name + ".geo");
-    std::string gmsh_command =
-        gmsh_binary + " -2 -algo meshadapt \"" + file_base_name + ".geo\"";
-    gmsh_command += " -o \"" + file_base_name + ".msh\"";
+    // fs::path const temp_dir = fs::temp_directory_path();
+    auto geo_file = fs::temp_directory_path() /= "createSurface.geo";
+    auto msh_file = fs::temp_directory_path() /= "createSurface.msh";
```

Comment:
The file name should be unique too. Otherwise running tests in parallel will cause problems.

Reply:
✅

Reply:
Unfortunately the generated names are not random enough. I run
`bin/testrunner --gtest_shuffle --gtest_repeat=10`
and it always hit an existing file name.
For some reason the files are also not cleaned up afterwards.

Reply:
This is now much better and works so far. Only few file were not cleaned up after testrunner finished successfully. I'll have a look tomorrow on this.

Reply:
I guess the main error was in `sizeof(charset) - 1`. Should be `sizeof(charset) - 2`: one for array count starting from 0, another one for the null terminator.